pallets/circuit/src/machine/test.rs

```diff
@@ -0,0 +1,238 @@
+#[cfg(test)]
+pub mod test {
```

Comment:
I like this stuff; looking so much cleaner already, wow. Nice with these guard tests, what I'm missing a bit is that we test every transition for correct but also incorrect transitions. Just to be sure that we cover everything, and no faulty transitions are possible. Maybe it makes sense to have a single test for each transition (e.g. kill) and then run the test from each available state, the allowed an forbidden ones.

The other thing would be to track an XTX through the different paths, and having them non-empty. But maybe that more for circuit tests?